mgmt_cli_test.py

```diff
@@ -871,6 +891,40 @@ def test_enospc_during_backup(self):
                     clean_enospc_on_node(target_node=target_node, sleep_time=30)
         self.log.info('finishing test_enospc_during_backup')
 
+    def test_enospc_before_restore(self):
+        self.log.info('starting test_enospc_before_restore')
+        manager_tool = mgmt.get_scylla_manager_tool(manager_node=self.monitors.nodes[0])
+        mgr_cluster = manager_tool.get_cluster(cluster_name=self.CLUSTER_NAME) \
+            or manager_tool.add_cluster(name=self.CLUSTER_NAME, db_cluster=self.db_cluster,
+                                        auth_token=self.monitors.mgmt_auth_token)
+        backup_task = mgr_cluster.create_backup_task(location_list=self.locations)
+        backup_task_status = backup_task.wait_and_get_final_status(timeout=1500)
+        assert backup_task_status == TaskStatus.DONE, \
+            f"Backup task ended in {backup_task_status} instead of {TaskStatus.DONE}"
+        target_node = self.db_cluster.nodes[1]
+        has_enospc_been_reached = False
+        with ignore_no_space_errors(node=target_node):
+            try:
+                reach_enospc_on_node(target_node=target_node)
+                has_enospc_been_reached = True
```

Comment:
There's a possible case where, for example, `reach_enospc_on_node` raise an exception after already getting to ENOSPC, but then it won't clean the node.

Reply:
You're right! No harm in cleaning either way. Fixing.

Reply:
Fixed, and also removed from the other test function that has used it